core/src/main/java/org/dotwebstack/framework/core/directives/ConstraintTraverser.java

```diff
@@ -0,0 +1,76 @@
+package org.dotwebstack.framework.core.directives;
+
+import static java.util.Optional.ofNullable;
+import static org.dotwebstack.framework.core.helpers.ObjectHelper.castToMap;
+
+import com.google.common.collect.ImmutableMap;
+import graphql.schema.DataFetchingEnvironment;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLDirective;
+import graphql.schema.GraphQLDirectiveContainer;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInputObjectField;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLScalarType;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@AllArgsConstructor
+public class ConstraintTraverser {
+
+  private final ConstraintValidator validator;
+
+  public void traverse(DataFetchingEnvironment dataFetchingEnvironment) {
+    GraphQLFieldDefinition fieldDefinition = dataFetchingEnvironment.getFieldDefinition();
+    Map<String,Object> arguments = dataFetchingEnvironment.getArguments();
+
+    fieldDefinition.getArguments().forEach(argument ->
+            onArguments(argument, arguments.get(argument.getName())));
+  }
+
+
+  void onArguments(GraphQLArgument argument, Object value) {
```

Comment:
zou simpeler kunnen:
```suggestion
    void onArguments(GraphQLArgument argument, Object value) {
    if (argument.getType() instanceof GraphQLInputObjectType) {
      onInputObjectType((GraphQLInputObjectType) argument.getType(),
              value != null ? castToMap(value) : ImmutableMap.of());
    } else  if ((argument.getType() instanceof GraphQLScalarType)) {
      validate(argument,argument.getName(),ofNullable(value).orElse(argument.getDefaultValue()));
    }
  }
```